Gulpfile.ts

```diff
@@ -0,0 +1,1125 @@
+    // This is silly. We include Q because orchestrator (a part of gulp) depends on it, but its not included.
+    // `del` further depends on `Promise` (and is also not included), so we just, patch the global scope's Promise to Q's
+    type Promise<T> = Q.Promise<T>;
+}
+import del = require("del");
+import mkdirP = require("mkdirP");
+import minimist = require("minimist");
+import browserify = require("browserify");
+import through2 = require("through2");
+import intoStream = require("into-stream");
+import * as os from "os";
+import Linter = require("tslint");
+const gulp = helpMaker(originalGulp);
+const mochaParallel = require("./scripts/mocha-parallel.js");
+const {runTestsInParallel} = mochaParallel;
+
+const cmdLineOptions = minimist(process.argv.slice(2), {
+    boolean: ["debug", "light", "colors", "lint", "soft"],
+    string: ["browser", "tests", "host", "reporter", ],
+    alias: {
+        d: "debug",
+        t: "tests",
+        test: "tests",
+        r: "reporter",
+        color: "colors",
+        f: "files",
+        file: "files"
+    },
+    default: {
+        soft: false,
+        colors: process.env.colors || process.env.color || true,
+        debug: process.env.debug || process.env.d,
+        host: process.env.TYPESCRIPT_HOST || process.env.host || "node",
+        browser: process.env.browser || process.env.b || "IE",
+        tests: process.env.test || process.env.tests || process.env.t,
+        light: process.env.light || false,
+        port: process.env.port || process.env.p || "8888",
+        reporter: process.env.reporter || process.env.r,
+        lint: process.env.lint || true,
+        files: process.env.f || process.env.file || process.env.files || "",
+    }
+});
+
+function exec(cmd: string, args: string[], complete: () => void = (() => {}), error: (e: any, status: number) => void = (() => {})) {
+    console.log(`${cmd} ${args.join(" ")}`);
+    const ex = cp.spawn(cmd, args);
+    ex.stdout.pipe(process.stdout);
+    ex.stderr.pipe(process.stderr);
+    ex.on("exit", (code) => code === 0 ? complete() : error(/*e*/ undefined, code));
+    ex.on("error", error);
+}
+
+
+let useDebugMode = true;
+let host = cmdLineOptions["host"];
+
+// Constants
+const compilerDirectory = "src/compiler/";
+const servicesDirectory = "src/services/";
+const serverDirectory = "src/server/";
+const harnessDirectory = "src/harness/";
+const libraryDirectory = "src/lib/";
+const scriptsDirectory = "scripts/";
+const unittestsDirectory = "tests/cases/unittests/";
+const docDirectory = "doc/";
+
+const builtDirectory = "built/";
+const builtLocalDirectory = "built/local/";
+const LKGDirectory = "lib/";
+
+const copyright = "CopyrightNotice.txt";
+
+const compilerFilename = "tsc.js";
+const LKGCompiler = path.join(LKGDirectory, compilerFilename);
+const builtLocalCompiler = path.join(builtLocalDirectory, compilerFilename);
+
+// add node_modules to path so we don"t need global modules, prefer the modules by adding them first

```

Comment:
This comment no longer accurately reflects what is happening in the following lines.